### PR TITLE
Like

### PR DESCRIPTION
**Pull request by Mandhularajitha**
added functionality to the liked,watchlater,history,playlist

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,20 +1,39 @@
 import "./App.css";
 import {Route,Routes} from "react-router-dom"
-import { Navbar } from "./Components/Navigation/Navbar";
-import { Sidebar } from "./Components/Sidebar/Sidebar";
-import { VideoCard } from "./Components/VideoCards/VideoCard";
 import MockmanEs from "mockman-js";
+import { Login } from "./Components";
+import { Signup } from "./Components/LoginSignUp/SignUp";
+import {VideoCard} from "./Components/VideoCards/VideoCard"
+import {NavContainer} from "./Components/NavRoutes/NavContainer"
```
Comment (shraddha15-creator) [inline src/App.js:7]: add index.js file in components folder, export all component in that index file and then import it here from index file

File: src/App.js [hunk 2]
```diff
@@ -1,20 +1,39 @@
...
+import { Liked } from "./Pages/Liked/Liked";
+import { PlayList } from "./Pages/PlayList/PlayList";
+import {WatchLater} from "./Pages/WatchLater/WatchLater";
+import { History } from "./Pages/History/History";
```
Comment (shraddha15-creator) [inline src/App.js:11]: same for pages as well

File: src/App.js [hunk 3]
```diff
@@ -1,20 +1,39 @@
...
-  {/* <Route path="/Sidebar" element={<Sidebar/>}/>  */}
-  {/* <Route path="/videoCard" element={<VideoCard/>}/> */}
-  <Route path="/mockman" element={<MockmanEs />}/>
+ 
+  <Route element={<NavContainer/>}>
+
+    <Route path="/" element={<VideoCard/>} />
+    <Route path="/login" element={<Login/>}/>
+    <Route path="/SignUp" element={<Signup />}/>
+    {/* <Route path="/Home" element={<Home/>}/> */}
```
Comment (shraddha15-creator) [inline src/App.js:27]: remove commented code

File: src/Components/Home/Home.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+  return (
+   <>
+   <section>
+        <div className="banner">
+            <div className="navbar">
+            </div>
+            
+            <div className="content">
+                <h1 className="title_container">Watch Programming vedios</h1>
```
Comment (shraddha15-creator) [inline src/Components/Home/Home.js:14]: typo in `videos ` spelling

File: src/Components/Home/Home.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+                <p>Watch more different Programming Vedios and improve Your Skills</p>
```
Comment (shraddha15-creator) [inline src/Components/Home/Home.js:15]: here also, it sholud be `**videos**`

File: src/Components/LoginSignUp/Login.js [hunk 6]
```diff
@@ -0,0 +1,65 @@
+import React, { useState } from "react";
+import { Link } from "react-router-dom";
+import { useAuth } from "../../Context/Authentication/LoginContext";
+
+// import "./Login.css"
```
Comment (shraddha15-creator) [inline src/Components/LoginSignUp/Login.js:5]: remove

File: src/Components/LoginSignUp/SignUp.js [hunk 7]
```diff
@@ -0,0 +1,88 @@
+import React, { useState } from "react";
+import { Link } from "react-router-dom";
+import { Navbar } from "../Navigation/Navbar";
+// import "./Login.css"
```
Comment (shraddha15-creator) [inline src/Components/LoginSignUp/SignUp.js:4]: remove

File: src/Components/LoginSignUp/SignUp.js [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+    password: "",
+    confirmPassword: "",
+  });
+
+  const submitHandler = (e) => {
+    e.preventDefault();
+    if (data.password === data.confirmPassword) {
+      console.log(data);
+    } else {
+      console.log("passwords are not same ");
```
Comment (shraddha15-creator) [inline src/Components/LoginSignUp/SignUp.js:22]: remove console logs

File: src/Components/LoginSignUp/SignUp.js [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+    }
+  };
+
+  return (
+    <>
+     <Navbar />
+    <div className="signup">
+      {console.log(data)}
```
Comment (shraddha15-creator) [inline src/Components/LoginSignUp/SignUp.js:30]: remove

File: src/Components/VideoCards/VideoCard.js [hunk 10]
```diff
@@ -1,52 +1,84 @@
 import React from "react";
 import "./VideoCard.css";
 import { videos } from "../../backend/db/videos";
+import { useLiked } from "../../Context/LikeContext";
+import { usePlayList } from "../../Context/PlayListContext";
+import { useWatchlater } from "../../Context/WatchLaterContext";
+import { useHistry } from "../../Context/HistryContext";
```
Comment (shraddha15-creator) [inline src/Components/VideoCards/VideoCard.js:7]: use index file in context folder also

File: src/Components/VideoCards/VideoCard.js [hunk 11]
```diff
@@ -1,52 +1,84 @@
...
+                      
                       <div className="author">
                         <img src={chennelProfile} alt="" />
                       </div>
                       <div className="title">
-                        <h3>
-                          {title}
-                        </h3>
+                        <h3>{title}</h3>
                         <a href="">{mechennelNa}</a>
```
Comment (shraddha15-creator) [inline src/Components/VideoCards/VideoCard.js:71]: what is {mechennelNa} here ?, did you mean {channelName}?

File: src/Context/Authentication/LoginContext.js [hunk 12]
```diff
@@ -0,0 +1,58 @@
+import axios from "axios";
+import React from "react";
+import { useState } from "react";
+import { createContext } from "react";
+import { useContext} from "react";
```
Comment (shraddha15-creator) [inline src/Context/Authentication/LoginContext.js:5]: import it like this: 

` import React, { useState, createContext, useContext } from "react"; `

File: src/Context/Authentication/LoginContext.js [hunk 13]
```diff
@@ -0,0 +1,58 @@
...
+  const [isAuth,setIsAuth] =  useState(authInitialState);
+  
+  const loginHandler = async (email, password) => {
+    try {
+      const response = await axios.post(`/api/auth/login`, {
+        email,
+        password,
+        
+      });
+      console.log(response,"cyjhiuikjkfdtykmo")
```
Comment (shraddha15-creator) [inline src/Context/Authentication/LoginContext.js:24]: remove console

File: src/Context/LikeContext.js [hunk 14]
```diff
@@ -0,0 +1,76 @@
...
+
+
+  const [likedState, likedDispatch] = useReducer(likeReducerFunc, {
+    likedVideo: [],
+  });
+
+
+const addToLike = async (video, likedDispatch) => {
+  
+    console.log(video,"shilu")
```
Comment (shraddha15-creator) [inline src/Context/LikeContext.js:20]: remove

File: src/Context/PlayListContext.js [hunk 15]
```diff
@@ -0,0 +1,115 @@
...
+      setUserList({ ...userList, title: "" });
+    } catch (error) {
+      console.log(error);
+    }
+  };
+
+  const addDataToList = (video, playListDispatch) => {
+    playListDispatch({ type: "SELECTED_ID", payload: video });
+    playListDispatch({ type: "MODAL", payload: true });
+    // playListDispatch({type:"MODAL_CLOSE",paylod:false})
```
Comment (shraddha15-creator) [inline src/Context/PlayListContext.js:94]: remove

File: src/Pages/PlayList/PlayList.js [hunk 16]
```diff
@@ -0,0 +1,88 @@
...
+import { usePlayList } from '../../Context/PlayListContext';
+// import { Navigate } from 'react-router-dom';
+const authtoken = localStorage.getItem("AuthToken");
+
+export const PlayList = () => {
+
+    const {playListState} = usePlayList();
+    const {playListDispatch,deletePlayList} = usePlayList();
+
+    console.log(playListState,"9090")
```
Comment (shraddha15-creator) [inline src/Pages/PlayList/PlayList.js:13]: remove

File: src/Pages/PlayList/PlayList.js [hunk 17]
```diff
@@ -0,0 +1,88 @@
...
+            },
+          });
+  
+          console.log("response", response.data.playlists);
+          playListDispatch({
+            type: "ADD_TO_PlayList",
+            payload: response.data.playlists,
+          });
+        } catch (error) {
+          console.log(error);
```
Comment (shraddha15-creator) [inline src/Pages/PlayList/PlayList.js:30]: you can use `console.error` for err handling

**Review by shraddha15-creator — APPROVED** [review]
Looks good
suggestions:
- Remove all console.log before pushing
- Keep the PR small

**Review by AnjaliDeshwani — APPROVED** [review]
Looks good! 
Suggestion- remove console.log before merging

File: src/App.js (hunk 1 above)
Comment (Mandhularajitha) [reply src/App.js:7]: ok i will do

File: src/App.js (hunk 3 above)
Comment (Mandhularajitha) [reply src/App.js:27]: ok

File: src/Components/Home/Home.js (hunk 4 above)
Comment (Mandhularajitha) [reply src/Components/Home/Home.js:14]: ok

File: src/Context/LikeContext.js (hunk 14 above)
Comment (Mandhularajitha) [reply src/Context/LikeContext.js:20]: ok

File: src/Pages/PlayList/PlayList.js (hunk 17 above)
Comment (Mandhularajitha) [reply src/Pages/PlayList/PlayList.js:30]: ok

**Pull request merged**